pkg/cloud/services/network/subnets.go

```diff
@@ -177,6 +178,14 @@ func (s *Service) reconcileSubnets() error {
 		return errors.New("expected at least 1 subnet but got 0")
 	}
 
+	// Reconciling the zone information for the subnets. Subnets are grouped
+	// by regular zones (availability zones) or edge zones (local zones or wavelength zones),
+	// based in the zone-type attribute for zone
+	if err := s.reconcileZoneInfo(subnets); err != nil {
+		record.Warnf(s.scope.InfraCluster(), "FailedNoZoneInfo", "Expected the zone attributes to be populated to subnet")
+		return errors.Wrapf(err, "expected the zone attributes to be populated to subnet")
+	}
+
```

Comment:
Considering the zoneInfo is set in the subnet creation ([lines 491+](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4901/files#diff-81cc1a3a0c87b48d3a080c89f13e6e2f71523bdf139d3e5ba363f0c04a08645bR492-R499) ): do we still need the reconciliation for those fields?

Thoughts @nrb @damdo @rvanderp3 ?

Reply:
Answer: yes. The BYO VPC/subnet (unmanaged) scenarios requires that flow since it will not reach the "create subnet".